sdk/test-utils/perf/test/event/mockEventReceiver2.ts

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { generateUuid } from "@azure/core-http";
+
+export interface Event {
+  body: string;
+}
+
+export interface SubscribeOptions {
+  /**
+   * The maximum number of partitions to receive from.
+   * - **Default**: `12`.
+   */
+  partitions?: number;
+  /**
+   * Raises the error after the specified time.
+   */
+  raiseErrorAfterInSeconds: number;
+  /**
+   * Max number of events a batch can have.
+   */
+  maxBatchSize: number;
+}
+
+/**
+ * The general event handler interface (used for streaming events).
+ */
+export interface EventHandlers {
+  /**
+   * Handler that processes event.
+   */
+  processEvents(event: Event[], context: { partitionId: number }): Promise<void>;
+  /**
+   * Handler that processes errors that occur during receiving.
+   */
+  processError(err: Error): Promise<void>;
+}
+
+export class MockEventReceiver2 {
```

Comment:
I think we should rename this to something better.  Maybe `MockEventHubConsumerClient` or similar, to make it clear this is designed to mock `EventHubConsumerClient`?

Reply:
Done